zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -102,12 +101,31 @@ impl<F: Field> ExecutionGadget<F> for CallGadget<F> {
         // Lookup values from stack
         cb.stack_pop(gas_word.expr());
         cb.stack_pop(callee_address_word.expr());
-        cb.stack_pop(value.expr());
```

Comment:
I think you can just do 
```
cb.condition(is_call.expr(), |cb| cb.stack_pop(value.expr()));
```
here, and then you don't need to change the lines after?

Reply:
I used `cb.condition(is_call.expr(), |cb| cb.stack_pop(value.expr()))` for this code before. But it seems that not work for me.

I found the `stack_pointer_offset` is just increased in the function `stack_pop` internal (not an expression). This causes the next stack pop/push has wrong `stack_pointer_offset`.

https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/zkevm-circuits/src/evm_circuit/util/constraint_builder.rs#L992

Reply:
I think that is a bug. I made a PR here to turn stack_pointer_offset into an expression so that `self.condition` can be used: https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/837.